docs/source/testing-and-debugging.md

```diff
@@ -43,71 +43,79 @@ from boa3.boa3 import Boa3
+test environment for smart contracts with a neo-go node. When installing boa-test-constructor, [neo-mamba](https://dojo.coz.io/neo3/mamba/index.html)
+will be installed too.
 
 ### Testing
 
-Before writing your tests, make sure you have a Neo-Express network for local tests.
-If you do not yet have a local network, open a terminal and run `neoxp create`.
-Please refer to [Neo-Express documentation](https://github.com/neo-project/neo-express/blob/master/docs/command-reference.md#neoxp-create)
-for more details of how to configure your local network. 
+Create a Python Script, import the `SmartContractTestCase` class, and create a test class that inherits it. To set up
+the test environment, you'll need to override the `setUpClass` method from `SmartContractTestCase`. This method is 
+synchronous, so if you need to set up asynchronous tasks, you can create another async method and use it int the 
+`asyncio.run` method from `asyncio`. Common operations would be: creating accounts, deploying the smart contract, 
+selecting your "main" smart contract, and transferring GAS to the new accounts.
 
-Create a Python Script, import the NeoTestRunner class, and define a function to test your smart contract. In this
-function you'll need a NeoTestRunner object, which takes the path of your Neo-Express network configuration file as an
-argument to set up the test environment.
+Then, create functions to test the expected behavior of your smart contract. To invoke or test invoke your smart 
+contract, use the `call` method from `SmartContractTestCase`. The two positional parameters are the name of the method 
+you want to invoke and a list of its arguments. The keyword parameters are the return type, a list of signing accounts,
+a list of signers, and the smart contract you want to invoke. If you get an error when calling a smart contract, then an
+error will be raised.
 
-You'll have to call the method `call_contract()` to interact with your smart contract. Its parameters are the path of
-the compiled smart contract, the smart contract's method, and the arguments if necessary. 
-This call doesn't return the result directly, but includes it in a queue of invocations. To execute all the invocations
-set up, call the method `execute()`. Then assert the result of your invoke to see if it's correct.
-
-Note that `invoke.result` won't be set if the execution fails, so you should also assert if `runner.vm_state` is valid 
-for your test case.
+To persist an invocation, use the `signing_accounts` parameter to pass a list of signing accounts when calling the 
+smart contract. If you don't pass it, then it will always be a test invoke. The `signers` parameter can be used 
+alongside the `signing_accounts` if you want to change the witness scope of the invocation, or by itself if you want to
+test invoke but also define the signers of the transaction.
```

Comment:
we are not going to explain what signers and witnesses scopes are in this file, can you link the neo documentation related to signers here to guide the user if it wants more information?